optaplanner-examples/src/test/java/org/optaplanner/examples/conferencescheduling/persistence/ConnectionFollowRedirectsTest.java

```diff
@@ -0,0 +1,116 @@
+import static org.mockserver.model.HttpResponse.response;
+
+public class ConnectionFollowRedirectsTest {
+
+    private final String host = "localhost";
+    private final int port = 1080;
+
+    private MockServerClient mockServerClient;
+    private ClientAndServer mockServer;
+
+    @Before
+    public void startServer() {
+        mockServer = startClientAndServer(port);
+    }
+
+    @After
+    public void stopServer() {
+        mockServer.stop();
+    }
+
+    @Test
+    public void shouldRespondWithoutRedirection() {
+        mockServerClient = new MockServerClient(host, port);
+        mockServerClient
+                .when(
+                        request()
+                                .withMethod("GET")
+                                .withPath("/path")
+                )
+                .respond(
+                        response()
+                                .withStatusCode(200)
+                );
+
+        try {
+            ConnectionFollowRedirects connectionFollowRedirects = new ConnectionFollowRedirects(getUrl("path"));
+            assertTrue(connectionFollowRedirects.getConnection() instanceof HttpURLConnection);
+            connectionFollowRedirects.getInputStream().close();
+            assertEquals(connectionFollowRedirects.getRedirects(), 0);
+            assertEquals(((HttpURLConnection) connectionFollowRedirects.getConnection()).getResponseCode(), 200);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void shouldRedirectHttpOnce() {
+        mockServerClient = new MockServerClient(host, port);
+        mockServerClient
+                .when(
+                        request()
+                                .withMethod("GET")
+                                .withPath("/path")
+                )
+                .respond(
+                        response()
+                                .withStatusCode(301)
+                                .withHeader("location", getUrl("anotherPath"))
+                );
+        mockServerClient
+                .when(
+                        request()
+                                .withMethod("GET")
+                                .withPath("/anotherPath")
+                )
+                .respond(
+                        response()
+                                .withStatusCode(200)
+                );
+
+        try {
+            ConnectionFollowRedirects connectionFollowRedirects = new ConnectionFollowRedirects(getUrl("path"));
+            assertTrue(connectionFollowRedirects.getConnection() instanceof HttpURLConnection);
+            connectionFollowRedirects.getInputStream().close();
+            assertEquals(connectionFollowRedirects.getRedirects(), 1);
+            assertEquals(((HttpURLConnection) connectionFollowRedirects.getConnection()).getResponseCode(), 200);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
e.printStackTrace as well as System.out.println cannot be easily logged into file. Please use either logging framework or throw an exception.